Assets/HoloToolkit/UX/Scripts/Buttons/Button.cs

```diff
@@ -0,0 +1,488 @@
+            if (eventData.InputSource.TryGetSourceKind(eventData.SourceId, out sourceInfo))
+            {
+                if (sourceInfo == InteractionSourceInfo.Hand)
+                {
+                    m_HandCount++;
+                    m_bHandVisible = true;
+                }
+            }
+        }
+
+        /// <summary>
+        ///  On Source lost decrement hand count and set visiblity
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            InteractionSourceInfo sourceInfo;
+            if (eventData.InputSource.TryGetSourceKind(eventData.SourceId, out sourceInfo))
+            {
+                if (sourceInfo == InteractionSourceInfo.Hand)
+                {
+                    m_HandCount--;
+                    m_bHandVisible = m_HandCount <= 0;
+                }
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Called when button is pressed down.
+        /// </summary>
+        protected void DoButtonPressed(bool bRelease = false)
+        {
+            ButtonStateEnum newState = ButtonStateEnum.Pressed;
+            this.OnStateChange(newState);
+
+            if (OnButtonPressed != null)
+            {
+                OnButtonPressed(gameObject);
+            }
+
+            if (bRelease)
+            {
+                StartCoroutine("DelayedRelease", 0.2f);
+            }
+        }
+
+        /// <summary>
+        /// Called when button is released.
+        /// </summary>
+        protected void DoButtonReleased()
+        {
+            ButtonStateEnum newState;
+
+            if(m_bFocused)
+            {
+                newState = m_bHandVisible ? ButtonStateEnum.Targeted : ButtonStateEnum.ObservationTargeted;
+            }
+            else
+            {
+                newState = m_bHandVisible ? ButtonStateEnum.Interactive : ButtonStateEnum.Observation;
+            }
+
+            this.OnStateChange(newState);
+
+            if (OnButtonReleased != null)
+            {
+                OnButtonReleased(gameObject);
+            }
+        }
+
+        /// <summary>
+        /// Delayed function to release button works for click events
+        /// </summary>
+        /// <param name="delay"></param>
+        /// <returns></returns>
+        private IEnumerator DelayedRelease(float delay)
+        {
+            yield return new WaitForSeconds(delay);
+            DoButtonReleased();
+        }
+
+        /// <summary>
+        /// Called while button is pressed down.
+        /// </summary>
+        protected void DoButtonHeld()
+        {
+            if (OnButtonHeld != null)
+            {
+                OnButtonHeld(gameObject);
+            }
+        }
+
+        /// <summary>
+        /// Called when something interrupts the button pressed state.
+        /// </summary>
+        protected void DoButtonCancelled()
```

Comment:
spelling: Canceled

Reply:
Yeah it appears cancelled has been a common misspelling along with interactible (that's my fault :)).  Batch replace throughout project.

thanks,
-pat